integreat_cms/static/src/js/pages/page-preview.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Generate a page preview pop-up.
+ */
+
+import tinymce from "tinymce";
+
+window.addEventListener("load", () => {
+  const overlay = document.getElementById("preview_overlay");
+
+  document.querySelectorAll("[data-preview-page]").forEach((btn) => {
+    btn.addEventListener("click", () => {openPreviewWindow(overlay, btn) });
+  });
+
+  document.getElementById("btn-close-preview")
+    .addEventListener("click", () => {closePreviewWindow(overlay)});
+
+  // close window by clicking on backdrop
+  overlay.addEventListener("click", (e) => {
+    if (e.target === overlay) {
+      closePreviewWindow(overlay);
+    }
+  });
+});
+
+
+export default function openPreviewWindow(overlay:HTMLElement, btn: Element) {
+  fetch(btn.getAttribute("data-preview-page"))
+    .then(response => response.json())
+    .then(async (data) => {
+      overlay.classList.remove("hidden");
+      overlay.classList.add("flex");
+
+      let title = data["title"];
+      let pageContent = data["page_translation"];
+      let mirroredPageContent = data["mirrored_translation"];
+      let mirroredPageFirst = data["mirrored_page_first"];
+
+      if (btn.hasAttribute("data-edit-page-mode")) {
+        title = (document.getElementById("id_title") as HTMLInputElement)?.value;
+        pageContent = tinymce.activeEditor.getContent();
+
+        // fetch mirrored page content based on the current dropdown value states
+        const mirroredPageRegion = data["region_slugs"][(document.getElementById("mirrored_page_region") as HTMLInputElement)?.value];
```

Comment:
This selector does not seem to work for me, I have selected a mirrored page from augsburg, but it tries to fetch the content from nurnberg...